master/internal/rm/kubernetesrm/kubernetes_resource_manager.go

```diff
@@ -55,6 +55,14 @@ func New(
 	if err != nil {
 		panic(errors.Wrap(err, "failed to set up TLS config"))
 	}
+
+	// TODO clusterID should just be a `internal/config` package singleton.
```

Comment:
plan on doing this todo and removing setClusterID in favor of that? i figure this is just to not have to handle the annoying/impossible error from GetOrCreate... all over

Reply:
I added a ticket for it, maybe I'll do it after this lands

spec.go doesn't return errors so it makes it somewhat hard 

FWIW the error isn't impossible we requery GetOrCreate everytime, also we pass clusterID down like a sauasage in like a ton of places where I don't think we probably should

Also blank clusterID in pod.go on restart, but I think the field goes unusued anyway
https://github.com/determined-ai/determined/blob/7c87ee865082ebeb3bbd16013017294077a29ae9/master/internal/rm/kubernetesrm/pods.go#L523-L533

Reply:
lol. yeah, let's add a singleton somewhere else eventually, config sounds ok but maybe makes for some weird circular imports.